apps/files/src/actions/deleteAction.ts

```diff
@@ -183,7 +186,19 @@ export const action = new FileAction({
 			return Promise.all(nodes.map(() => false))
 		}
 
-		return Promise.all(nodes.map(node => this.exec(node, view, dir)))
+		// Map each node to a promise that resolves with the result of exec(node)
+		const promises = nodes.map(node => {
+		    // Create a promise that resolves with the result of exec(node)
+		    const promise = new Promise<boolean>(resolve => {
+				queue.add(async () => {
+					const result = await this.exec(node, view, dir)
+					resolve(result !== null ? result : false)
+				})
+			})
+			return promise
+		})
```

Comment:
```suggestion
		const promises = nodes.map(node => queue.add(() => this.exec(node, view, dir)))
```
This should work the same, didn't test

Also null is valid as a file action return type

Reply:
That was the first thing I tried. It didn't work.

The issue with that code snippet is related to the usage of `queue.add` within `Promise.all`

`queue.add` method returns a promise that resolves when the queued function is executed, but it doesn't return the actual result of the function. Therefore, using `queue.add` directly inside `Promise.all` will not work as intended because `Promise.all` expects an array of promises that resolve with the result of each executed function.

I didn't run it at all, but before I could have the chance to run it my code editor was complaining already.

Reply:
Update : Even though my IDE type/linting config does not seem to like it, it works, so implemented. Thanks.

Reply:
Hey @Pytal CI does not like this too : https://github.com/nextcloud/server/actions/runs/9076246755/job/24938632270?pr=45237

I guess I have to revert.

Reply:
Maybe:
`const promises = nodes.map(node => queue.add(() => this.exec(node, view, dir)) ?? null)`

Reply:
Still leading to type issues.